File: Dynamic-Programming/96-M-Unique-Binary-Search-Trees/unique_bst.cpp
class Solution {
public:
    int factorial (int n) {
        if (n <= 1)
            return 1;
        return factorial(n-1)*n;
    }

    int numTrees(int n) {
        if (n <= 1)
            return 1;
        return factorial(2*n)/(factorial(n)*factorial(n+1));    
    }
};

File: Dynamic-Programming/96-M-Unique-Binary-Search-Trees/unique_bst.py
class Solution:
    def factorial(self, n: int) -> int:
        if n <= 1:
            return 1
        return self.factorial(n-1)*n

    def numTrees(self, n: int) -> int:
        if n <= 1:
            return 1
        return self.factorial(2*n)//(self.factorial(n)*self.factorial(n+1))